src/test/test-copy.c

```diff
@@ -50,6 +51,135 @@ TEST(copy_file) {
         unlink(fn_copy);
 }
 
+TEST(copy_tree_replace_file) {
+        _cleanup_free_ char *buf = NULL;
+        char src[] = "/tmp/test-copy_file.XXXXXX";
+        char dest[] = "/tmp/test-copy_file.XXXXXX";
+
+        size_t sz = 0;
+        int fd;
+
+        fd = mkostemp_safe(src);
+        assert_se(fd >= 0);
+        close(fd);
+
+        fd = mkostemp_safe(dest);
+        assert_se(fd >= 0);
+        close(fd);
+
+        assert_se(write_string_file(src, "bar bar", WRITE_STRING_FILE_CREATE) == 0);
+        assert_se(write_string_file(dest, "foo foo foo", WRITE_STRING_FILE_CREATE) == 0);
+
+        assert_se(read_full_file(dest, &buf, &sz) == 0);
+        assert_se(streq(buf, "foo foo foo\n"));
+        assert_se(sz == 12);
+
+        // The file has been copied once - now overwite original contents, and test the COPY_REPLACE flag.
+
+        assert_se(copy_tree(src, dest, UID_INVALID, GID_INVALID, COPY_REFLINK) != 0);
```

Comment:
`!= 0`?

usually one would do `< 0` to check for errors or '>= 0` to check for success...